.github/actions/deploy-ads/action.yml

```diff
@@ -0,0 +1,169 @@
+      if: ${{ inputs.deploy-dir-network == 'true' }}
+      id: boostrap-node
+      shell: bash
+      env:
+        DIRECTORY_LOGGER_LOG_LEVEL: ERROR
+        DIRECTORY_IMAGE_TAG: ${{ inputs.directory-image-tag }}
+        DIRECTORY_CHART_TAG: ${{ inputs.directory-chart-tag }}
+        KIND_CLUSTER_NAME: ${{ inputs.kind-cluster-name }}
+        NETWORK_NAMESPACE_PREFIX: ${{ inputs.network-namespace-prefix }}
+        DIRCTL_BIN_VERSION: ${{ inputs.dirctl-bin-version }}
+        GH_TOKEN: ${{ github.token }}
+        ARCH: ${{ runner.arch }}
+        OS: ${{ runner.os }}
+      run: |
+        kubectl config use-context kind-$KIND_CLUSTER_NAME
+        cat <<EOF > /tmp/bootstrap.yaml
+          apiserver:
+            config:
+              # Routing settings for the peer-to-peer network.
+              routing:
+                # Address to use for routing
+                listen_address: '/ip4/0.0.0.0/tcp/8999'
+
+                # Path to private key file for peer ID.
+                key_path: /etc/agntcy/dir/node.privkey
+
+                # Nodes to use for bootstrapping of the DHT.
+                # We read initial routing tables here and get introduced
+                # to the network.
+                bootstrap_peers: []
+        EOF
+        echo "downloading dirctl-${OS}-${ARCH} ..."
+        gh release download ${DIRCTL_BIN_VERSION} --repo agntcy/dir --pattern "dirctl-${OS}-${ARCH}" --skip-existing --output /tmp/dirctl
+        chmod +x /tmp/dirctl
+        test -f /tmp/node.privkey || openssl genpkey -algorithm ED25519 -out /tmp/node.privkey
+        bootstrap_peerid=$(/tmp/dirctl network info /tmp/node.privkey)
+        echo "PEER ID: ${bootstrap_peerid}"
+        echo BOOTSTRAP_PEER_ID="${bootstrap_peerid}" >> $GITHUB_OUTPUT
+        helm upgrade agntcy-dir \
+          oci://ghcr.io/agntcy/dir/helm-charts/dir \
+          --version $DIRECTORY_CHART_TAG \
+          --set apiserver.image.tag="${DIRECTORY_IMAGE_TAG}" \
+          --set apiserver.privKey="$(cat /tmp/node.privkey)" \
+          --set apiserver.config.oci.registry_address="agntcy-dir-zot:5000" \
+          -f /tmp/bootstrap.yaml \
+          --namespace "${NETWORK_NAMESPACE_PREFIX}-bootstrap" \
+          --create-namespace \
+          --install \
+          --wait \
+          --wait-for-jobs \
+          --timeout "15m"
+
+    - name: deploy agntcy ADS network nodes
+      if: ${{ inputs.deploy-dir-network == 'true' }}
+      shell: bash
+      env:
+        BOOSTRAP_PEER_ID: ${{ steps.boostrap-node.outputs.BOOTSTRAP_PEER_ID }}
+        DIRECTORY_LOGGER_LOG_LEVEL: ERROR
+        DIRECTORY_IMAGE_TAG: ${{ inputs.directory-image-tag }}
+        DIRECTORY_CHART_TAG: ${{ inputs.directory-chart-tag }}
+        KIND_CLUSTER_NAME: ${{ inputs.kind-cluster-name }}
+        NETWORK_NAMESPACE_PREFIX: ${{ inputs.network-namespace-prefix }}
+      run: |
+        kubectl config use-context kind-$KIND_CLUSTER_NAME
+        peers=("peer1" "peer2" "peer3")
```

Comment:
can we make this configurable as well?

Reply:
doing

Reply:
I've added `network-size`: Number of directory network peers (default: `3`)